src/lib/Scenes/Artwork/Components/ImageCarousel/ImageWithLoadingState.tsx

```diff
@@ -18,74 +20,65 @@ interface ImageWithLoadingStateProps {
-  useEffect(
-    () => {
-      Animated.spring(overlayOpacity, {
-        toValue: isLoading ? 1 : 0,
-        useNativeDriver: true,
-      }).start()
-    },
-    [isLoading]
-  )
+  const overlayOpacity = useSpringValue(isLoading ? 1 : 0)
 
   // show a loading spinner only after a short delay, if the image is taking a while to load
-  const spinnerOpacity = useMemo(() => new Animated.Value(0), [])
+  const [showingSpinner, setShowingSpinner] = useState(false)
+  const spinnerOpacity = useSpringValue(showingSpinner ? 1 : 0)
   useEffect(() => {
-    setTimeout(() => {
-      Animated.spring(spinnerOpacity, {
-        toValue: 1,
-        useNativeDriver: true,
-      }).start()
-    }, 1000)
+    setTimeout(() => setShowingSpinner(true), 1000)
   }, [])
-  const { width, height, imageURL } = props
+  const { width, height, imageURL, onPress } = props
   return (
-    <View style={[{ width, height }, props.style]}>
-      <View style={{ position: "absolute", width, height }}>
-        <OpaqueImageView
-          onLoad={() => {
-            setIsLoading(false)
-            if (props.onLoad) {
-              props.onLoad()
-            }
-          }}
-          imageURL={imageURL}
-          aspectRatio={width / height}
-          style={{ width, height }}
-        />
-      </View>
-      <Animated.View
-        style={{
-          position: "absolute",
-          width,
-          height,
-          opacity: overlayOpacity,
-          backgroundColor: "white",
-          flex: 1,
-        }}
-      >
+    <TouchableWithoutFeedback onPress={onPress}>
+      <View style={[{ width, height }, props.style]} ref={ref}>
+        <View style={{ position: "absolute", width, height }}>
+          <OpaqueImageView
+            noAnimation
+            disableGemini
+            onLoad={() => {
+              setIsLoading(false)
+              if (props.onLoad) {
+                props.onLoad()
+              }
+            }}
+            imageURL={imageURL}
+            aspectRatio={width / height}
+            style={{ width, height }}
+          />
+        </View>
         <Animated.View
           style={{
-            opacity: spinnerOpacity,
-            // give the image a subtle silhouette while the spinner is displaying
-            // to keep the balance of the page and set an appropriate user expectation
-            backgroundColor: color("black5"),
-            alignItems: "center",
-            justifyContent: "center",
+            position: "absolute",
+            width,
+            height,
+            opacity: overlayOpacity,
+            backgroundColor: "white",
             flex: 1,
           }}
         >
-          <Spinner />
+          <Animated.View
+            style={{
+              opacity: spinnerOpacity,
+              // give the image a subtle silhouette while the spinner is displaying
+              // to keep the balance of the page and set an appropriate user expectation
```

Comment:
Thanks for all of the code-level comments / documentation left throughout this work, it's very helpful 👍